### PR TITLE
fix: sorting by non-existing value throws `INVALID_SERVER_ERROR` on Postgres

### LINKED ISSUE #8072: Sorting by non-existing field returns http error 500
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
If I delete a field that was previously sorted/ordered in Parse Dashboard, when I try to load the records from that class, returns a 500 error.

### Steps to reproduce
I created a column (named ‘s’) in the Contact class to test the schema, then I deleted it via Dashboard and via JS SDK but when I click on the dashboard Contact Class, no records are showing, and parse server log complains with error: 
```bash
error: column "s" does not exist {"code":1,"stack":"Error: error: column \"s\" does not exist\n    at /Users/Martin/Dev/Parse-Server/node_modules/parse-server/lib/Controllers/DatabaseController.js:1164:21\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"}
```
If I create the field again (via dashboard or via JS) the error goes away and the records appear

I found the cause in the POST request when I  select the Contact Class that returns a 500 error
```
XHRPOST http://localhost:1337/parse/classes/Contact
[HTTP/1.1 500 Internal Server Error 6ms]
_ApplicationId	"appId"
_ClientVersion	"js3.4.1"
_InstallationId	"8ad5475f-30e5-4a1"
_MasterKey	"masterKey"
_method	"GET"
limit	200
order	"s"  //<-- here is where I realize the problem
where	{}
```

### Actual Outcome
Error 500

### Expected Outcome
Not get the error and load the records

### Environment
Local environment

Server
- Parse Server version: `5.3.2`
- Operating system: `macOS Monterrey`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Local`

Database
- System (MongoDB or Postgres): `Postgres`
- Database version: `14.3`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Local` (I did not tried in remote)

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `JavaScript`
- SDK version: `3.4.2`

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Sorting by a non-existent string throws on postgres.

Closes #8072

### Approach

Checks if the sort key exists in `SCHEMA`

### TODOs before merging

- [x] Add tests
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -1207,6 +1207,9 @@ class DatabaseController {
                 `Invalid field name: ${fieldName}.`
               );
             }
+            if (!schema.fields[fieldName.split('.')[0]]) {
```
Comment (mtrezza) [inline src/Controllers/DatabaseController.js]: What should be the expected outcome when sorting by non-existing field? Ignoring the field or throwing a 400 "invalid field name" error? I would infer that from how Parse Server behaves in similar situations, e.g. when using a non-existing field in an `equalTo` condition.

Comment (dblythy) [reply src/Controllers/DatabaseController.js]: I think it should just delete that sort key as a safe failure, but we can throw here if you think it's a better approach

Comment (mtrezza) [reply src/Controllers/DatabaseController.js]: How does Parse Server behave in similar situations, e.g. when using a non-existing field in an equalTo condition?

Comment (dblythy) [reply src/Controllers/DatabaseController.js]: It succeeds and returns nothing, e.g:

```js
const query = new Parse.Query('a').equalTo('b','c');
const obj = await query.first()
// obj is undefined
```

Comment (mtrezza) [reply src/Controllers/DatabaseController.js]: Interesting, in which cases is the error "invalid key name" used then?

Comment (dblythy) [reply src/Controllers/DatabaseController.js]: I think that's mostly used to protect internal / sensitive keys (prevent sorting by authdata for example)

Comment (mtrezza) [reply src/Controllers/DatabaseController.js]: Well, if query conditions fail gracefully, then I think sort should do the same. So we can keep the approach as is.

**Comment by mtrezza** [comment]
Since you've made another change, please re-request a review when this is ready.

**Comment by dblythy** [comment]
Ready for review, not sure how to request on mobile

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.26](https://github.com/parse-community/parse-server/releases/tag/5.3.0-alpha.26)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)

**Pull request merged**